lib/Zonemaster/Backend/TestAgent.pm

```diff
@@ -170,9 +173,10 @@ sub run {
         }
     }
 
-    $self->{_db}->test_results( $test_id, Zonemaster::Engine->logger->json( 'INFO' ) );
+    my @entries = grep { $_->numeric_level >= $numeric{INFO} } @{ Zonemaster::Engine->logger->entries };
```

Comment:
Did you consider filtering the entries before we even buffer them, as opposed to before we store them? I'm thinking we could save some memory that way.